From 24a926ea3caf56d72158c5d577d0c7c059567f33 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Sat, 7 Mar 2020 13:48:50 +0100
Subject: [PATCH] types.h: Fix ambiguous reference to weak_ptr

Without this patch the build fails with:

	error: reference to 'weak_ptr' is ambiguous

As there also seems to be a definition of weak_ptr in the std namespace.
---
 types.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/types.h b/types.h
index be9fb1f6..a5747d05 100644
--- a/types.h
+++ b/types.h
@@ -346,13 +346,13 @@ struct enable_weak_from_this {
         }
     }
 
-    weak_ptr<T> weak() { return weak_ptr<T>(static_cast<T*>(this)); }
+    ::weak_ptr<T> weak() { return ::weak_ptr<T>(static_cast<T*>(this)); }
 
     void schedule_deletion() {
         fdevent_run_on_looper([this]() { delete static_cast<T*>(this); });
     }
 
   private:
-    friend struct weak_ptr<T>;
-    std::vector<weak_ptr<T>*> weak_ptrs_;
+    friend struct ::weak_ptr<T>;
+    std::vector<::weak_ptr<T>*> weak_ptrs_;
 };
